src/main/java/com/appdirect/sdk/vendorFields/controller/VendorRequiredFieldController.java

```diff
@@ -118,4 +125,10 @@ public void initBinder(final WebDataBinder webdataBinder) {
         webdataBinder.registerCustomEditor(List.class, new LocaleConverter());
         webdataBinder.registerCustomEditor(Locale.class, new LocaleObjectConverter());
     }
+
+    private Map<String, String> getAdditionalInformation(Map<String, String> headers) {
+        return new HashMap<String, String>() {{
+                put(HttpHeaders.X_FORWARDED_FOR, headers.get(HttpHeaders.X_FORWARDED_FOR));
```

Comment:
I think we can rename getAdditionalInformation to be specific to this use case to explain why we're propagating the XFF header.

I understand you want it to be easy to add more headers in the future but YAGNI comes into play.

Alternatively, we could do something like this (the method name could be whatever you think makes more sense):
```suggestion
                put(addSourceIP());
```

Reply:
I don't think YAGNI is really applicable here as this is some minor syntax issue and YAGNI is more on whether it is worth spending time on over engineering stuff

Reply:
Either way, we should make it more readable than just "getAdditionalInformation"

Reply:
I think about it as the idea of what the function is trying to achieve. The purpose of the function is to add additional mappings and pass it to the connector. If I am adding a string type to the connector I would be using getSourceIP/getXFFHeader. But I am trying to avoid doing that because we dont want to add too many additional parameters in the future. And we dont want to refactor those retrospectively because it will mean breaking the API